### DISCUSSION TITLE
Creating a note from a template that has a saved name and title of the folder name plus appended text

### DISCUSSION THREAD

**Discussion by NetDevAutomate** [SilentVoid13/Templater#1171] [Help]
I'm tying to create a note from a template that has a saved name and title of the folder name plus appended text.

An example is to 'Create note from template' in a folder with structure 2-Areas/Code/Python

I want to name the note Python Tasks and maintain the metadata in the template:

```
---
title: <% tp.file.folder() %>
date: <% moment(tp.file.creation_date()).format("YYYY-MM-DD HH:mm:ss") %>
creation_date: <% tp.file.creation_date() %>
modification_date: <% tp.file.last_modified_date("dddd, Do MMMM YYYY, HH:mm:ss") %>
lastmod
Project: <% tp.file.folder() %>
categories:
tags: taskbone todoist
aliases:
---
```

I've tried everything I can think of... but not matter where I perform the rename, I loose all of the existing content of the template.

I've tried using a myriad of other examples people have kindly published but no luck, so hoping this is the right place to ask for guidance. 

One example of the template:

```
---
title: <% tp.file.folder()+" Tasks" %>
date: <% moment(tp.file.creation_date()).format("YYYY-MM-DD HH:mm:ss") %>
creation_date: <% tp.file.creation_date() %>
modification_date: <% tp.file.last_modified_date("dddd, Do MMMM YYYY, HH:mm:ss") %>
Project: <% tp.file.folder() %>
categories:
tags: taskbone todoist
aliases:
---

 <% await tp.file.rename(tp.file.folder()+" Tasks")%>
#  <% tp.file.title %>`
```

- Without the tp.file.rename : I get the correct content but with an "Untitled" name
- With rename I have the correct file name but no content (including metadata)

Any guidance would be grateful appreciated.

**Comment by Zachatoo** [comment]
The reason why `tp.file.title` results in `Undefined` is because when you rename the file, `tp.file.title` is not updated to the new file name, it is set when the template begins execution and is not updated after the rename.

I modified your script to declare a `title` variable (using an [execution command](https://silentvoid13.github.io/Templater/commands/execution-command.html)) at the top of the file to be referenced for the rename and the heading. I also added [whitespace control](https://silentvoid13.github.io/Templater/commands/whitespace-control.html) to ensure that when the template is executed, the frontmatter will still be on the top line of the file and be properly parsed by Obsidian as frontmatter.

```js
<%*
const title = tp.file.folder()+" Tasks";
await tp.file.rename(title);
-%>
---
title: <% title %>
date: <% moment(tp.file.creation_date()).format("YYYY-MM-DD HH:mm:ss") %>
creation_date: <% tp.file.creation_date() %>
modification_date: <% tp.file.last_modified_date("dddd, Do MMMM YYYY, HH:mm:ss") %>
Project: <% tp.file.folder() %>
categories:
tags: taskbone todoist
aliases:
---

#  <% title %>
```

**Reply by NetDevAutomate** [reply]
@Zachatoo Firstly, thank you SO much for such a quick response, and secondly for a perfect solution. I am incredibly grateful as it works perfectly.

You've also given enough of an explanation for me to know where to spend time to get up to speed with Templater.

I'm new to Obsidian and Javascript, but starting my TypeScript journey so hope to be a get way more involved with Templater.

Again, my sincere thanks.
